core/src/crds_gossip_pull.rs

```diff
@@ -508,7 +511,10 @@ impl CrdsGossipPull {
                             Some(item.value.clone())
                         }
                     })
-                    .collect()
+                    .take(output_size_limit)
+                    .collect();
+                output_size_limit -= out.len();
```

Comment:
Should we restructure this to early exit out of the iterator when `output_size_limit` hits zero to avoid doing the rest of the filters?

Reply:
done